From 4f32efe70b618c4721bf197692fc9d709d352528 Mon Sep 17 00:00:00 2001
From: Kienan Stewart <kstewart@efficios.com>
Date: Mon, 20 Nov 2023 11:27:12 -0500
Subject: [PATCH 1/2] fix: phys_proc_id and cpu_core_id moved in linux
 6.7.0-rc1

See upstream commit:

    commit 02fb601d27a7abf60d52b21bdf5b100a8d63da3f
    Author: Thomas Gleixner <tglx@linutronix.de>
    Date:   Mon Aug 14 10:18:30 2023 +0200

        x86/cpu: Move phys_proc_id into topology info

        Rename it to pkg_id which is the terminology used in the kernel.

        No functional change.

See upstream commit:

    commit e95256335d45cc965cd12c423535002974313340
    Author: Thomas Gleixner <tglx@linutronix.de>
    Date:   Mon Aug 14 10:18:34 2023 +0200

        x86/cpu: Move cpu_core_id into topology info

        Rename it to core_id and stick it to the other ID fields.

        No functional change.

Upstream-Status: Backport [commit 4f32efe7]

Signed-off-by: Kienan Stewart <kstewart@efficios.com>
Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Change-Id: I574b02430210d5bb72c4b9db901d0e3a6dc7bea0
---
 .../instrumentation/events/lttng-statedump.h  | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/include/instrumentation/events/lttng-statedump.h b/include/instrumentation/events/lttng-statedump.h
index 642aa356..4831a941 100644
--- a/include/instrumentation/events/lttng-statedump.h
+++ b/include/instrumentation/events/lttng-statedump.h
@@ -253,6 +253,23 @@ LTTNG_TRACEPOINT_EVENT(lttng_statedump_interrupt,
 
 #define LTTNG_HAVE_STATEDUMP_CPU_TOPOLOGY
 
+#if (LTTNG_LINUX_VERSION_CODE >= LTTNG_KERNEL_VERSION(6,7,0))
+LTTNG_TRACEPOINT_EVENT(lttng_statedump_cpu_topology,
+	TP_PROTO(struct lttng_kernel_session *session, struct cpuinfo_x86 *c),
+	TP_ARGS(session, c),
+	TP_FIELDS(
+		ctf_string(architecture, "x86")
+		ctf_integer(uint16_t, cpu_id, c->cpu_index)
+		ctf_string(vendor, c->x86_vendor_id[0] ? c->x86_vendor_id : "unknown")
+		ctf_integer(uint8_t, family, c->x86)
+		ctf_integer(uint8_t, model, c->x86_model)
+		ctf_string(model_name, c->x86_model_id[0] ? c->x86_model_id : "unknown")
+		ctf_integer(uint16_t, physical_id, c->topo.pkg_id)
+		ctf_integer(uint16_t, core_id, c->topo.core_id)
+		ctf_integer(uint16_t, cores, c->booted_cores)
+	)
+)
+#else
 LTTNG_TRACEPOINT_EVENT(lttng_statedump_cpu_topology,
 	TP_PROTO(struct lttng_kernel_session *session, struct cpuinfo_x86 *c),
 	TP_ARGS(session, c),
@@ -268,6 +285,8 @@ LTTNG_TRACEPOINT_EVENT(lttng_statedump_cpu_topology,
 		ctf_integer(uint16_t, cores, c->booted_cores)
 	)
 )
+#endif /* LTTNG_LINUX_VERSION_CODE >= LTTNG_KERNEL_VERSION(6,7,0) */
+
 #endif /* CONFIG_X86_32 || CONFIG_X86_64 */
 
 #endif /*  LTTNG_TRACE_LTTNG_STATEDUMP_H */
-- 
2.34.1

